### PR TITLE
Support copying span contexts.

### PR DESCRIPTION
**Pull request by rnburn**
Adds support for copying span contexts.

Follow up from https://gitter.im/opentracing/opentracing-cpp?at=5a82810018f388e62697ab54

### CODE REVIEW

**Comment by rnburn** [comment]
@isaachier - Not a high priority, but could you take a look at this when you get the chance?

File: mocktracer/src/mock_span_context.cpp [hunk 1]
```diff
@@ -21,12 +21,20 @@ void MockSpanContext::ForeachBaggageItem(
   }
 }
 
-void MockSpanContext::SetData(SpanContextData& data) {
+void MockSpanContext::SetData(SpanContextData& data) const {
```
Comment (isaachier) [inline mocktracer/src/mock_span_context.cpp]: This name is a bit confusing. Sounds like you are setting the data member named `data`. Maybe `CopyData`?

Comment (rnburn) [reply mocktracer/src/mock_span_context.cpp]: Sure. I'll rename it.

**Comment by d4l3k** [comment]
@rnburn Do you know when this will make it into a tagged release? It'd be really nice to be able to use this

**Pull request merged**